app.py

```diff
@@ -0,0 +1,120 @@
+        'exportDataAccessGroups': 'false',
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+
+    if len(response.json()) == 0:
+        return None
+
+    assert len(response.json()) == 1, \
+        f"Error: Multiple records matching {filter_logic}"
+
+    user_data = response.json()[0]
+    return {
+        'record_id': user_data['record_id'],
+        'eligibility_screening_complete': user_data['eligibility_screening_complete']
+    }
+
+def register_net_id(net_id: str) -> str:
+    """
+    Returns the REDCap record ID of the participant newly registered under the
+    given *net_id*.
+    """
+    # REDCap enforces that we must provide a non-empty record ID. Because we're
+    # using `forceAutoNumber` in the POST request, we do not need to provide a
+    # real record ID.
+    values = [{'netid': net_id, 'record_id': 'record ID cannot be blank'}]
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'overwriteBehavior': 'normal',
+        'forceAutoNumber': 'true',
+        'data': json.dumps(values),
+        'returnContent': 'ids',
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.json()[0]
+
+
+def generate_survey_link(record_id: str) -> str:
+    """
+    Returns a generated survey link to the eligibility screening instrument for
+    the given *record_id*.
+    """
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'surveyLink',
+        'format': 'json',
+        'instrument': 'eligibility_screening',
+        'event': 'enrollment_arm_1',  # TODO subject to change
+        'record': record_id,
+        'returnFormat': 'json'
+    }
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.text
+
+
+@app.route('/')
+def main():
+    # Get NetID from Shibboleth data
+    net_id = request.remote_user
+
+    while not net_id:
+        # TODO: Redirect to Shibboleth login
+        net_id = 'KaasenG'
+        pass
+
+    user_data = fetch_user_data(net_id)
+
+    if user_data is None:
+        # If not in REDCap project, create new record
+        user_data = {'record_id': register_net_id(net_id)}
+
+    # TODO -- generate a survey link for a particular day
+    # We are awaiting finalization of the REDCap project to know how
+    # daily attestations (repeating instruments) will be implemented.
+    if is_complete('eligibility_screening', user_data):
+        return f"Congrats, {net_id}, you're already registered under record ID " \
+            f"{user_data['record_id']} and your eligibility " \
+            "screening is complete!"
+
+    # Generate a link to the eligibility questionnaire, and then redirect
+    return redirect(generate_survey_link(user_data['record_id']))
```

Comment:
Do we also want to check if `consent_form` and `enrollment_questionnaire` are complete and redirect participants to those survey links if they are not complete? I think the participant is only officially enrolled in the study after completing the `enrollment_questionnaire`.

Reply:
Via a new commit, users are redirected to REDCap if their eligibility screening, consent form, or enrollment questionnaire is incomplete.